src/CoreWCF.MSMQ/tests/Helpers/DependencyResolverHelper.cs

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace CoreWCF.MSMQ.Tests.Helpers
+{
+    internal class DependencyResolverHelper
+    {
+        private readonly IWebHost _webHost;
+
+        public DependencyResolverHelper(IWebHost webHost)
+        {
+            _webHost = webHost;
+        }
+
+        public T GetService<T>()
+        {
```

Comment:
Potentially fragile. I don't know if a scoped ServiceProvider has a finalizer registered, but if it does, it could dispose anything it created.